app/concerns/long_runnable.rb

```diff
@@ -0,0 +1,80 @@
+module LongRunnable
+  extend ActiveSupport::Concern
+
+  included do |base|
+    AgentRunner.register(base)
+  end
+
+  def start_worker?
+    true
+  end
+
+  def worker_id(config = nil)
+    "#{self.class.to_s}-#{id}-#{Digest::SHA1.hexdigest((config.presence || options).to_json)}"
+  end
+
+  module ClassMethods
+    def setup_worker
+      active.map do |agent|
+        next unless agent.start_worker?
+        self::Worker.new(id: agent.worker_id, agent: agent)
+      end.compact
+    end
+  end
+
+  class Worker
```

Comment:
I added some basic documentation, since I am pretty bad at writing docs feedback is very welcome :smile: